base/datatypes.py

```diff
@@ -286,7 +286,7 @@ def setSweepParameter(self, name, start, end, step, step_type):
 
     value_range = []
     if step_type == "linstep":
-      value_range = range(start, end+1, step)
+      value_range = [i for i in range(start, end + 1, step)]
```

Comment:
I believe you can just cast this to a list:

```
list(range(start, end+1, step))
```

Reply:
> I believe you can just cast this to a list:
> 
> ```
> list(range(start, end+1, step))
> ```

Right. Done.